src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/StrandArtifact.java

```diff
@@ -0,0 +1,178 @@
+import htsjdk.variant.variantcontext.GenotypeBuilder;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFormatHeaderLine;
+import htsjdk.variant.vcf.VCFHeaderLineType;
+import org.apache.commons.math3.analysis.integration.gauss.GaussIntegrator;
+import org.apache.commons.math3.analysis.integration.gauss.GaussIntegratorFactory;
+import org.apache.commons.math3.distribution.BetaDistribution;
+import org.apache.commons.math3.distribution.BinomialDistribution;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.utils.GATKProtectedVariantContextUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Annotations for strand artifact filter
+ *
+ * Created by tsato on 4/19/17.
+ */
+public class StrandArtifact extends GenotypeAnnotation implements StandardSomaticAnnotation {
+    public static final String POSTERIOR_PROBABILITIES_KEY = "SA_POST_PROB";
+    public static final String MAP_ALLELE_FRACTIONS_KEY = "SA_MAP_AF";
+
+    // pseudocounts for the beta distribution over epsilon
+    // alpha > 0 and beta > 0. alpha = beta = 1 gives us the flat prior
+    // give more prior weight to beta (i.e. pseudocount for tails) so that the peak shifts towards 0
+    final int ALPHA = 2;
+    final int BETA = 6;
+
+    // the latent variable z in the strand artifact filter model
+    public enum StrandArtifactZ {
+        ART_FWD, ART_REV, NO_ARTIFACT;
+    }
+
+    @Override
+    public List<String> getKeyNames() {
+        return Arrays.asList(POSTERIOR_PROBABILITIES_KEY, MAP_ALLELE_FRACTIONS_KEY);
+    }
+
+    @Override
+    public void annotate(final ReferenceContext ref,
+                         final VariantContext vc,
+                         final Genotype g,
+                         final GenotypeBuilder gb,
+                         final ReadLikelihoods<Allele> likelihoods) {
+        Utils.nonNull(gb);
+        Utils.nonNull(vc);
+        Utils.nonNull(likelihoods);
+
+        // do not annotate the genotype fields for normal
+        if (g.isHomRef()){
+            return;
+        }
+
+        final double[] pi = new double[]{ 0.025, 0.025, 0.95 };
+
+        // We use the allele with highest LOD score
+        final double[] tumorLods = GATKProtectedVariantContextUtils.getAttributeAsDoubleArray(vc, GATKVCFConstants.TUMOR_LOD_KEY, () -> null, -1);
+        final int indexOfMaxTumorLod = MathUtils.maxElementIndex(tumorLods);
+        final Allele altAlelle =  vc.getAlternateAllele(indexOfMaxTumorLod); // Careful!
+
+        final Collection<ReadLikelihoods<Allele>.BestAllele> tumorBestAlleles = likelihoods.bestAlleles();
```

Comment:
But we skip the normal genotype, so the BestAllele is always for tumor. Do you think it's still better to give it a more general name?